contest/scoreboard.py

```diff
@@ -0,0 +1,104 @@
+'''
+    The MIT License (MIT)
+    Copyright (c) 2014 NTHUOJ team
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    '''
+import sys
+class Scoreboard:
+    def __init__(self,start_time):
+        self.users = []
+        self.problems = []
+        self.start_time = start_time
+    def add_user(self,user):
+        self.users.append(user)
+    def add_problem(self,problem):
+        self.problems.append(problem)
+    def get_problem(self,pname):
+        for problem in self.problems:
+            if (problem.pname == pname):
+                return problem
+    
+class User:
+    def __init__(self,username):
+        self.username = username
+        self.problems = []
```

Comment:
Can you access problem list in scoreboard?
It doesn't make sense to keep entire problem list in each user object.